pkg/api/apiloader.go

```diff
@@ -205,6 +205,9 @@ func (a *Apiloader) LoadContainerService(
 		if containerService.Properties == nil {
 			return nil, errors.New("missing ContainerService Properties")
 		}
+		if containerService.Properties.IsAzureStackCloud() && containerService.Location == "" {
```

Comment:
Let's move this validation into `pkg/api/vlabs/validate.go`

Reply:
The other validations in this surface are tech debt, we will (not in this PR) move those into `validate.go` as well

Reply:
I also put the following code into validate.go
		if containerService.Properties == nil {
			return nil, errors.New("missing ContainerService Properties")
		}